### PR TITLE
Brush up README.md

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Will self-review now.

**Review by elbrujohalcon — APPROVED** [review]
Looking great so far.

File: README.md [hunk 1]
```diff
@@ -1,85 +1,86 @@
-# elvis-core [![Build Status](https://github.com/inaka/elvis_core/workflows/build/badge.svg)](https://github.com/inaka/elvis_core) [![Stories in Ready](https://badge.waffle.io/inaka/elvis.png?label=ready&title=Ready)](https://waffle.io/inaka/elvis)
```
Comment (paulo-ferraz-oliveira) [inline README.md:1]: Broken link "Stories in Ready" removed

File: README.md [hunk 2]
```diff
@@ -1,85 +1,86 @@
...
+[ci]: https://github.com/inaka/elvis_core
+[ci-img]: https://github.com/inaka/elvis_core/workflows/build/badge.svg
 
-This library includes the mechanism to apply rules to your code and their
-implementation.
+`elvis_core` is the core library for the [`elvis`](https://github.com/inaka/elvis) Erlang style
+reviewer. It is also used by [`rebar3_lint`](https://github.com/project-fifo/rebar3_lint) for easier
+integration into your Erlang libraries or applications.
 
-## Contributing & Reporting Bugs
```
Comment (paulo-ferraz-oliveira) [inline README.md:8]: Section moved to the end of the document

File: README.md [hunk 3]
```diff
@@ -1,85 +1,86 @@
...
+# elvis_core [![GitHub Actions CI][ci-img]][ci]
 
-Erlang style reviewer core library.
+[ci]: https://github.com/inaka/elvis_core
+[ci-img]: https://github.com/inaka/elvis_core/workflows/build/badge.svg
 
-This library includes the mechanism to apply rules to your code and their
-implementation.
+`elvis_core` is the core library for the [`elvis`](https://github.com/inaka/elvis) Erlang style
+reviewer. It is also used by [`rebar3_lint`](https://github.com/project-fifo/rebar3_lint) for easier
```
Comment (paulo-ferraz-oliveira) [inline README.md:7]: Added mention to `rebar3_lint`. Hope that's Ok

File: README.md [hunk 4]
```diff
@@ -1,85 +1,86 @@
...
-The [elvis](https://github.com/inaka/elvis) repository is where all elvis_core
-related work is coordinated.
+It implements [pre-defined rules](#pre-defined-rules), but also supports
+[user-defined ones](#user-defined-rules).
 
 ## Usage
 
 ### As a library
 
-This library implements all the core functionality for analyzing files and
```
Comment (paulo-ferraz-oliveira) [inline README.md:21]: Removed since it basically repeats what was said before

File: README.md [hunk 5]
```diff
@@ -1,85 +1,86 @@
...
-extensively, so please check that project for a concrete example on how
-you could use for your own purposes.
-
-### Erlang Shell
+The [`elvis`](https://github.com/inaka/elvis) command-line tool uses `elvis_core` extensively, so
+do check that project for a concrete example on how you could use it for your own purposes.
 
-After adding `elvis_core` as a dependency to your project and starting a
-shell, you will need to make sure the application is started:
+### As a `rebar3` plugin
```
Comment (paulo-ferraz-oliveira) [inline README.md:22]: Again `rebar3_lint`

File: README.md [hunk 6]
```diff
@@ -1,85 +1,86 @@
...
+2> elvis_core:rock(ElvisConfig).
+# src/elvis_core.erl [OK]
+# src/elvis_result.erl [OK]
+# src/elvis_style.erl [OK]
+# src/elvis_utils.erl [OK]
+ok
+3>
 ```
 
-**IMPORTANT**: `Config` should have a valid format, but since this is a project
```
Comment (paulo-ferraz-oliveira) [inline README.md:71]: Removed notion of "under development". This is a production-ready plugin already

File: README.md [hunk 7]
```diff
@@ -93,163 +94,166 @@ rules:
...
 
-### Files, Rules & Rulesets
+### Files, rules and rulesets
 
-The `dirs` key is a list that indicates where `elvis` should look for the
-files that match `filter`, which will be run through each of the default rules
-in the specified `ruleset`, which is an *atom*.
+The `dirs` key is a list that tells `elvis_core` where it should look for the files that match
+`filter`, which will be run through each of the pre-defined rules in the specified `ruleset`.
+`filter` can contain `**` for further matching (it uses
```
Comment (paulo-ferraz-oliveira) [inline README.md:131]: Added this bit (`**`) as I feel it's important to mention here

File: README.md [hunk 8]
```diff
@@ -93,163 +94,166 @@ rules:
...
+[`filelib:wildcard/1`](https://erlang.org/doc/man/filelib.html#wildcard-1) under the hood).
 
-If you want to override the [default rules][default-rules] for a given ruleset
-you need to specify them in a `rules` key which is a list of items with the
-following structure `{Module, Function, RuleConfig}` or `{Module, Function}`
-if the rule takes no configuration values. You can also `disable` certain rules
-if you want to just by specifying the rule in the `rules` key and passing
-`disable` as its third parameter.
+If you want to override the [pre-defined rules](#pre-defined-rules), for a given ruleset, you need
```
Comment (paulo-ferraz-oliveira) [inline README.md:134]: I renamed (it's a conceptual change only) "default" rules as "pre-defined" rules and "custom" rules as "user-defined" rules. This is more visible in a section further below

File: README.md [hunk 9]
```diff
@@ -93,163 +94,166 @@ rules:
...
-  ```
+```erlang
+#{ dirs => ["src"]
+ , filter => "*.erl"
+ , ruleset => erl_files
+ , ignore => [module1, module4]
+}.
+```
 
-  With this configuration, none of the checks for [erl_files](https://github.com/inaka/elvis_core/blob/master/src/elvis_rulesets.erl#L6-L34) would be applied to `module1.erl` and `module4.erl` files.
```
Comment (paulo-ferraz-oliveira) [inline README.md:188]: Removed broken reference to code

File: README.md [hunk 10]
```diff
@@ -93,163 +94,166 @@ rules:
...
-analyzed; and a configuration map specified for the rule. This means you can
-define rules of your own as long as the functions that implement them respect
-this arity.
+### User-defined rules
 
-## References
+The implementation of a new rule is a function that takes 3 arguments in the following order:
 
-Inspired by [HoundCI][houndci]
+1. `elvis_config:config()`: the value of option `config` as found in the
```
Comment (paulo-ferraz-oliveira) [inline README.md:243]: I tweaked this so it's easier to follow (the user-defined rules' interface)

**Comment by paulo-ferraz-oliveira** [comment]
Added a bunch of comments. Do review them or Resolve if you're Ok with the changes. Moving to Ready for review.

**Comment by paulo-ferraz-oliveira** [comment]
Thanks, Brujo.

**Pull request merged**